src/sentry/static/sentry/app/utils/bulkController.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import xor from 'lodash/xor';
+import uniq from 'lodash/uniq';
+
+export type RenderProps = {
+  allIdsSelected: boolean;
+  pageIdsSelected: boolean;
+  onIdToggle: (id: string) => void;
+  onAllIdsToggle: (select: boolean) => void;
+  onPageIdsToggle: (select: boolean) => void;
+} & Pick<State, 'selectedIds'>;
+
+type State = {
+  /**
+   * Currently selected ids across pages
+   */
+  selectedIds: string[];
+};
+
+type Props = {
+  /**
+   * Array of all ids across pagination pages
+   */
+  allIds: string[];
```

Comment:
Do we often have all the ids available in addition to the ids on the current page?